pom.xml

```diff
@@ -515,7 +490,7 @@
                 </requirePluginVersions>
                 -->
                 <enforceBytecodeVersion>
-                  <maxJdkVersion>1.${java.level}</maxJdkVersion>
+                  <maxJdkVersion>${java.level}</maxJdkVersion>
```

Comment:
Good thing this PR explicitly drops support for `java.level=8`, then, as stated in the PR description:

> Drop support for Java 8

Reply:
Yeah but we cannot just do that because then people cannot simply accept updates to the parent POM without also updating `jenkins.version` to something very new.

Reply:
I mean, we could, but then we have plugin POM versions that work with older cores but are not well suited to newer cores, and POM versions that require newer cores, without any overlap. So this seems pretty drastic when we normally encourage people to update the parent even when the core version is a few LTS lines behind. For example, anyone staying with an older line would not be able to accept new POM versions for totally unrelated reasons (new tools or JTH).

If we really go with the flag day approach in this PR, then we should definitely get rid of the `java.level` property as it serves no purpose. Add a profile making it an error to set it, and hard-code 11 as the Java level. And make sure that the build fails early and clearly when your `jenkins.version` is too old.

Reply:
I think we can. We can always do a backport branch if we want to deliver plugin parent POM updates to plugins that do not want to upgrade `jenkins.version` to something very new.

Reply:
> So this seems pretty drastic

Yes it is drastic but nevertheless feasible without changing the current semantics of `java.level` being defined in a `<properties>` section in plugin `pom.xml` files. Other proposals like #133 are not feasible as far as I can tell so long as feedback like [this](https://github.com/jenkinsci/plugin-pom/pull/133#discussion_r359344017) remain unaddressed.

> Instead figure out what Java level to pass to tools based on the jenkins.version (somehow).

That approach could be feasible (and less drastic!), but it would be a lot more work and (as of today) I am explicitly not volunteering to do that work.

Reply:
Agree with Basil, because of how this was done you can't keep compat.

We can run a 4.x branch if we need to backport changes...

alternatively plugins that want to stay on old core could just set this to 1.8?

Reply:
> set this to 1.8

IIRC that will break _other_ places which expect `8`.